httpbp/client_middlewares.go

```diff
@@ -349,3 +354,43 @@ func PrometheusClientMetrics(serverSlug string) ClientMiddleware {
 		})
 	}
 }
+
+func FaultInjection() ClientMiddleware {
+	return func(next http.RoundTripper) http.RoundTripper {
+		return roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			resumeFn := func() (*http.Response, error) {
+				return next.RoundTrip(req)
+			}
+			responseFn := func(code int, message string) (*http.Response, error) {
+				return &http.Response{
```

Comment:
I'm not sure constructing `*http.Response` by hand like this is a good idea. cc @pacejackson as you wrote the majority of the httpbp implementation 😅

Reply:
I didn't write the any of the `Client` though 😅

Reply:
I think that was @konradreiche

Reply:
Yeah, I originally wanted to emulate a server a closely as possible via `ResponseWriter`, but it's not designed to be used in a way that would allow me to not _actually_ send something and just extract the constructed raw `http.Response` from it. This was the next best thing that I could determine, though let me know if you know of a less manual solution.

Reply:
It looks like thats [what ResponseRecorder does](https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/net/http/httptest/recorder.go;l=188) so I would imagine it's fine?l